libs/libarchfpga/src/physical_types_util.cpp

```diff
@@ -836,28 +816,7 @@ std::tuple<const t_sub_tile*, int> get_sub_tile_from_class_physical_num(t_physic
     return std::make_tuple(nullptr, -1);
 }
 
-t_logical_block_type_ptr get_logical_block_from_class_physical_num(t_physical_tile_type_ptr physical_tile, int physical_class_num) {
-    VTR_ASSERT(physical_class_num >= (int)physical_tile->class_inf.size());
-
-    const t_sub_tile* sub_tile;
-    int sub_tile_cap;
-    std::tie(sub_tile, sub_tile_cap) = get_sub_tile_from_class_physical_num(physical_tile, physical_class_num);
-    VTR_ASSERT(sub_tile_cap != -1);
-    int class_num_offset = get_sub_tile_inst_physical_class_num_offset(physical_tile, sub_tile, sub_tile_cap);
-    VTR_ASSERT(physical_class_num >= class_num_offset);
-
-    for (auto tmp_logical_block : sub_tile->equivalent_sites) {
-        if (physical_class_num < (class_num_offset + (int)tmp_logical_block->logical_class_inf.size())) {
-            return tmp_logical_block;
-        } else {
-            class_num_offset += (int)tmp_logical_block->logical_class_inf.size();
-        }
-    }
-
-    return nullptr;
-}
-
-std::vector<int> get_pin_list_from_class_physical_num(t_physical_tile_type_ptr physical_tile, int class_physical_num) {
+const std::vector<int>& get_pin_list_from_class_physical_num(t_physical_tile_type_ptr physical_tile, int class_physical_num) {
```

Comment:
Add a comment to explain the inputs and outputs of the function and why you need to return a const